RoboticonColony/Assets/Code/Market.cs

```diff
@@ -1,132 +1,220 @@
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                //If the item transfer was unsuccessful, then revert the money transfer and re-throw the exception
+                Stock.TransferMoney(_buyprice[item] * quantity, playerInventory);
+                throw;
+            }
+        }
+        catch (ArgumentOutOfRangeException)
+        {
+            //If the initial money transfer was unsuccessful, then re-throw the exception
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Allows a player to sell to the market.
+    /// </summary>
+    /// <param name="item">The item the player wishes to sell.</param>
+    /// <param name="quantity">The quantity the player wishes to sell</param>
+    /// <param name="playerInventory">Reference to the players inventory.</param>
+    /// <returns>This market reference, for method chaining.</returns>
+    public Market Sell(ItemType item, int quantity, Inventory playerInventory)
+    {
+        //Attempt to transfer money from the market to the player.
+        try
+        {
+            Stock.TransferMoney(_sellprice[item] * quantity, playerInventory);
+
+            //Attempt to transfer the requested item(s) into the markets inventory.
+            try
+            {
+                playerInventory.TransferItem(item, quantity, Stock);
+
+                //If the transfer completes without error, then the transaction is complete and a reference to this market instance is returned.
+                return this;
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                //If the item transfer was unsuccessful, then revert the money transfer and re-throw the exception
+                playerInventory.TransferMoney(_sellprice[item] * quantity, Stock);
+                throw;
+            }
+        }
+        catch (ArgumentOutOfRangeException)
+        {
+            //If the initial money transfer was unsuccessful, then re-throw the exception
+            throw;
+
+        }
+    }
+
+    /// <summary>
+    /// customisationsList is a list of all RoboticonCustomisations
+    /// </summary>
+    public List<RoboticonCustomisation> customisationsList { set; private get; }
+
+    /// <summary>
+    /// Creates an customisation type for roboticons.
+    /// </summary>
+    /// <param name="selectedName"> The name of the customisation. </param>
+    /// <param name="bonusMult"> The multiplier which the production will be boosted by. </param>
+    /// <param name="prereq"> The list of other customisations which must completed already before this customisation can be applied. </param>
+    /// <param name="selectedResource"> The type of resource which the customisation augments. </param>
+    /// <param name="reqPrice"> The required price of the customisation. </param>
+    private void createCustomisations(string selectedName, int bonusMult, List<RoboticonCustomisation> prereq, ItemType selectedResource, int reqPrice)

```

Comment:
This is here so that when there are more customisations, like reading from a file, the function can be called rather than 20 lines of code in initialization